### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by skabou**
Image cleaner no longer in preview, now GA
AKS 1.29
Traefik 2.11
AzureLinux
bugfix: container image was missing curl
bugfix: kustomization patchesJson6902->patches

end-to-end
curl

### CODE REVIEW

File: 07-bootstrap-validation.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ GitOps allows a team to author Kubernetes manifest files, persist them in their
 - Cluster-wide configuration of Azure Monitor for Containers
 - The workload's namespace named `a0008`
 
-1. Install `kubectl` 1.27 or newer. (`kubectl` supports ±1 Kubernetes version.)
+1. Install `kubectl` 1.28 or newer. (`kubectl` supports ±1 Kubernetes version.)
```
Comment (ferantivero) [inline 07-bootstrap-validation.md:13]: are we maintaining the client version n-1 for a reason? Any desire to move over 1.29 instead?

File: 10-workload.md [hunk 2]
```diff
@@ -47,7 +47,7 @@ The cluster now has an [Traefik configured with a TLS certificate](./09-secret-m
    > You should expect a `403` HTTP response from your ingress controller if you attempt to connect to it *without* going through the App Gateway. Likewise, if any workload other than the ingress controller attempts to reach the workload, the traffic will be denied via network policies.
 
    ```bash
-   kubectl run curl -n a0008 -i --tty --rm --image=mcr.microsoft.com/azure-cli --overrides='[{"op":"add","path":"/spec/containers/0/resources","value":{"limits":{"cpu":"200m","memory":"128Mi"}}},{"op":"add","path":"/spec/containers/0/securityContext","value":{"readOnlyRootFilesystem": true}}]' --override-type json  --env="DOMAIN_NAME=${DOMAIN_NAME_AKS_BASELINE}"
+   kubectl run curl -n a0008 -i --tty --rm --image=mcr.microsoft.com/devcontainers/base --overrides='[{"op":"add","path":"/spec/containers/0/resources","value":{"limits":{"cpu":"200m","memory":"128Mi"}}},{"op":"add","path":"/spec/containers/0/securityContext","value":{"readOnlyRootFilesystem": true}}]' --override-type json  --env="DOMAIN_NAME=${DOMAIN_NAME_AKS_BASELINE}"
```
Comment (ferantivero) [inline 10-workload.md:50]: nice catch and proposal change 🎉

File: nested_K8sCustomIngressTlsHostsHaveDefinedDomainSuffix.bicep [hunk 3]
```diff
@@ -84,57 +84,6 @@ resource pdK8sCustomIngressTlsHostsHaveDefinedDomainSuffix 'Microsoft.Authorizat
         }
         defaultValue: {
         }
-        schema: {
```
Comment (ferantivero) [inline nested_K8sCustomIngressTlsHostsHaveDefinedDomainSuffix.bicep:87]: have we considered aligning the schema to [this example](https://learn.microsoft.com/en-us/azure/governance/policy/concepts/definition-structure#example-2)?

File: workload/aspnetapp-ingress-patch.yaml [hunk 4]
```diff
@@ -3,4 +3,4 @@
   value: bu0001a0008-00.aks-ingress.contoso.com
 - op: replace
   path: "/spec/rules/0/host"
-  value: bu0001a0008-00.aks-ingress.contoso.com
```
Comment (ferantivero) [inline workload/aspnetapp-ingress-patch.yaml:6]: consider excluding this file from the PR and all others just changing chars.

**Review by ferantivero — CHANGES_REQUESTED** [review]
nicely done @skabou, please find some feedback below.

File: 07-bootstrap-validation.md (hunk 1 above)
Comment (skabou) [reply 07-bootstrap-validation.md:13]: That was the guidance I received from @ckittel on a previous PR. 

Also, it appears to be [official policy from Kubernetes](https://kubernetes.io/releases/version-skew-policy/#kubectl)

File: nested_K8sCustomIngressTlsHostsHaveDefinedDomainSuffix.bicep (hunk 3 above)
Comment (skabou) [reply nested_K8sCustomIngressTlsHostsHaveDefinedDomainSuffix.bicep:87]: Thanks, I will investigate.  The deployment had been giving a warning about the "schema" key.

File: 07-bootstrap-validation.md (hunk 1 above)
Comment (ferantivero) [reply 07-bootstrap-validation.md:13]: Yeah, while Im aware of how it gets supported, I was wondering why we are prescribing that. Just wanted to learn from this.

File: nested_K8sCustomIngressTlsHostsHaveDefinedDomainSuffix.bicep (hunk 3 above)
Comment (skabou) [reply nested_K8sCustomIngressTlsHostsHaveDefinedDomainSuffix.bicep:87]: (retaining for reference)
image

Comment (skabou) [reply nested_K8sCustomIngressTlsHostsHaveDefinedDomainSuffix.bicep]: Going to remove this file to address later

**Review by ferantivero — APPROVED** [review]
w00t @skabou, LGTM 🚢

**Pull request merged**